llvm/lib/Passes/PassBuilderPipelines.cpp

```diff
@@ -644,6 +653,9 @@ PassBuilder::buildFunctionSimplificationPipeline(OptimizationLevel Level,
   LPM1.addPass(LICMPass(PTO.LicmMssaOptCap, PTO.LicmMssaNoAccForPromotionCap,
                         /*AllowSpeculation=*/false));
 
+  if (EnableLoopIterCountToAssumptions)
+    LPM1.addPass(LoopIterCountAssumptions());
```

Comment:
Could we potentially run the pass twice?

Reply:
We run the pass and then immediately rotate the loop. 

So on the second entry, we already have a rotated loop and bail